core/src/main/java/io/questdb/griffin/SqlCompilerImpl.java

```diff
@@ -3125,7 +3125,11 @@ private void insertAsSelect(ExecutionModel executionModel, SqlExecutionContext e
                         }
                     } catch (Throwable e) {
                         // rollback data when system error occurs
-                        writer.rollback();
+                        try {
+                            writer.rollback();
+                        } catch (Throwable e2) {
+                            // Writer is distressed, exception already logged, the pool will handle it when writer is returned
```

Comment:
we should probably log our assumption here?

Reply:
ok, done